packages/image_picker/image_picker_ios/example/ios/RunnerTests/ImageUtilTests.m

```diff
@@ -13,26 +13,82 @@ @interface ImageUtilTests : XCTestCase
 
 @implementation ImageUtilTests
 
+static NSString *ColorStringAtPixel(UIImage *image, int pixelX, int pixelY) {
+  CGImageRef cgImage = image.CGImage;
+
+  uint32_t argb;
+  CGContextRef context1 = CGBitmapContextCreate(
+      &argb, 1, 1, CGImageGetBitsPerComponent(cgImage), CGImageGetBytesPerRow(cgImage),
+      CGColorSpaceCreateDeviceRGB(), CGImageGetBitmapInfo(cgImage));
+  CGContextDrawImage(
+      context1, CGRectMake(-pixelX, -pixelY, CGImageGetWidth(cgImage), CGImageGetHeight(cgImage)),
+      cgImage);
+  CGContextRelease(context1);
+  int blue = argb & 0xff;
+  int green = argb >> 8 & 0xff;
+  int red = argb >> 16 & 0xff;
+  int alpha = argb >> 24 & 0xff;
+
+  return [CIColor colorWithRed:red / 255.f
+                         green:green / 255.f
+                          blue:blue / 255.f
+                         alpha:alpha / 255.f]
+      .stringRepresentation;
+}
+
+- (void)testScaledImage_EqualSizeReturnsSameImage {
+  UIImage *image = [UIImage imageWithData:ImagePickerTestImages.JPGTestData];
+  UIImage *scaledImage = [FLTImagePickerImageUtil scaledImage:image
+                                                     maxWidth:@(image.size.width)
+                                                    maxHeight:@(image.size.height)
+                                          isMetadataAvailable:YES];
+
+  // Assert the same bytes pointer (not just equal objects).
+  XCTAssertEqual(image, scaledImage);
+}
+
 - (void)testScaledImage_ShouldBeScaled {
   UIImage *image = [UIImage imageWithData:ImagePickerTestImages.JPGTestData];
-  UIImage *newImage = [FLTImagePickerImageUtil scaledImage:image
-                                                  maxWidth:@3
-                                                 maxHeight:@2
-                                       isMetadataAvailable:YES];
 
-  XCTAssertEqual(newImage.size.width, 3);
-  XCTAssertEqual(newImage.size.height, 2);
+  CGFloat scaledWidth = 3;
+  CGFloat scaledHeight = 2;
+  UIImage *scaledImage = [FLTImagePickerImageUtil scaledImage:image
+                                                     maxWidth:@(scaledWidth)
+                                                    maxHeight:@(scaledHeight)
+                                          isMetadataAvailable:YES];
+  XCTAssertEqual(scaledImage.size.width, scaledWidth);
+  XCTAssertEqual(scaledImage.size.height, scaledHeight);
+
+  // Check the corners. Format is "R G B A"
+  XCTAssertEqualObjects(ColorStringAtPixel(scaledImage, 0, 0), @"1 0.776471 0 1");  // yellow
```

Comment:
Nit: How about making these color string representations consts with descriptive names so they can be reused and seem less like a magic value?

Reply:
I can do that for the two tests that are the same.  These exact values would changes based on how small the image is scaled.